File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        
        String answer = "";
        HashMap<String, Integer> map1 = new HashMap();
        for(String s : participant) {
            map1.put(s, map1.getOrDefault(s, 0) + 1);
        }
        
        HashMap<String, Integer> map2 = new HashMap();
        for(String s : completion) {
            map2.put(s, map2.getOrDefault(s, 0) + 1);
        }
        
        for(String s : participant) {
            if(!map2.containsKey(s)) {
                answer = s;
            } else if(map1.get(s) > map2.get(s)) {
                answer = s;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    tmp = {}
    answer = ""

    for a in participant:
        tmp[a] = tmp.get(a, 0) + 1
    
    
    for b in completion:
        if tmp.get(b) > 1:
            tmp[b] = tmp[b] - 1
        else:
            del tmp[b]
    
    for key in tmp.keys():
        answer = key
        
    return answer